scheduler/src/cook/kubernetes/api.clj

```diff
@@ -157,16 +172,28 @@
                                           (.getItems current-nodes))]
     (reset! current-nodes-atom node-name->node)
     (let [watch (WatchHelper/createNodeWatch api-client (-> current-nodes .getMetadata .getResourceVersion))]
-      (.submit kubernetes-executor ^Callable
       (fn []
         (try
           (handle-watch-updates current-nodes-atom watch (fn [n] (-> n .getMetadata .getName))
                                 [(make-atom-updater current-nodes-atom)]) ; Update the set of all nodes.
           (catch Exception e
-            (log/warn e "Error during node watch"))
+            (log/warn e "Error during node watch for compute cluster" compute-cluster-name))
           (finally
             (.close watch)
-            (initialize-node-watch api-client current-nodes-atom))))))))
+            (initialize-node-watch api-client compute-cluster-name current-nodes-atom)))))))
+
+(defn initialize-node-watch
+  "Initialize the node watch. This fills current-nodes-atom with data and invokes the callback on pod changes."
+  [^ApiClient api-client compute-cluster-name current-nodes-atom]
+    ; We'll iterate trying to connect to k8s until the initilzie-pod-watch-helper returns a watch function.
+  (let [tmpfn (fn [] (try
+                       (initialize-node-watch-helper api-client compute-cluster-name current-nodes-atom)
+                       (catch Exception e
+                         (log/error e "Error during initial setup of looking at nodes for" compute-cluster-name)
+                         (Thread/sleep 60000)
+                         nil)))
+        ^Callable first-success (->> tmpfn repeatedly (some identity))]
+    (.submit kubernetes-executor ^Callable first-success)))
```

Comment:
What happens if this line throws? Wouldn't we stop trying to re-start the watch?

Reply:
Same response as earlier.